### ISSUE TITLE
Settings Section order

### ISSUE THREAD

**Issue by ChunkyProgrammer** [FreeTubeApp/FreeTube#1739]
**Topic**
Please select what topic your discussion is about.
- [ ] Coding
- [ ] Contribution
- [ ] Languages
- [ ] Setup
- [x] Other. Please specify: UI

**Discussion**
Please write down whatever you want to discuss.
I don't believe that the order that setting headers are presented in is optimal
![Image](https://user-images.githubusercontent.com/84899178/133312832-9f64ad64-43bb-4f07-9bbc-8be747994070.png)

I feel like something like General then Theme then do the rest in alphabetical order would be better but the alphabetical order would change for different langauges so I don't think this would be ideal. Another option would be to save the order of the settings and allow user's to drag the headings to customize them.  I think a more accessible and easier way to implement this would be to just add arrows going up or down to the headers when a button called "Customize Headers" is clicked. Thoughts?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Yeah i agree that sorting it alphabetically is not feasible. I do like the second suggestion tho. It would be really nice if the user can select where they want the setting to be.

**Comment by jasonhenriquez** [comment]
I haven't had a problem with this personally, but I do think this kind of permanent customization of list-like elements could be useful.  Here are a few related issues: #1717 #987 #958 #17. Sorting by recency, sorting alphabetically, sorting manually, and "hiding" some selections seem to be all of the suggestions. If you did want to end up applying this capability to so many different kinds of lists (profiles, subscriptions, dropdown settings, app settings), you could add some or all of these capabilities to certain lists by adding a button near them that would open a settings modal.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Not sure if it belongs here and wanted to open up an issue for this but i thought maybe i can ask it here. If u think this doesnt belong here and i should or should not open an issue about it pls let me know.

Hiding everything related to invidious when invidious is not selected as API. I mean there are people that dont even use that api and it maybe uses to much screen real estate. So why show me if im never ever going to do anything with it? Its just unnecessary in my opinion.
![Screenshot (245)](https://user-images.githubusercontent.com/73130443/136676232-c4611ad6-bde7-452a-99ec-182552246f2a.png)

Same here why show all of the related settings when its not even enabled?
![Screenshot (246)](https://user-images.githubusercontent.com/73130443/136676233-ac7f2df9-dcdc-4893-ad38-fd0a0729cbaa.png)

**Comment by jasonhenriquez** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc This is partially addressed in #1822. In particular, the Proxy Settings are now grayed when "Enable Tor/Proxy" is disabled. Are you still of the opinion that it should just disappear instead of being grayed out? Similarly, are you still of the opinion that "Current invidious instance" should disappear when "Fallback to Non-Preferred Backend on Failure" is set to false & Preferred API Backend is set to "Local API"? (This second case is not grayed out in #1822 at this time.)

My concern is that removing them instead of just graying them could cause confusion for people unfamiliar with that interaction who are looking for that functionality.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
As discussed in IRC, graying out will be just fine!
